### PR TITLE
Add gcc build dependency to pear

### PR DESCRIPTION
**Pull request by marcelm**
- [x] I have read the guidelines above.
- [ ] This PR adds a new recipe.
- [x] This PR updates an existing recipe.
- [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/pear/meta.yaml [hunk 1]
```diff
@@ -1,22 +1,28 @@
...
-  number: 1
+  number: 2
+
 requirements:
   build:
+    - gcc
     - pkgconfig
     - zlib
   run:
     - zlib
```
Comment (marcelm) [inline recipes/pear/meta.yaml]: I’ll add it. But [the error](https://travis-ci.org/bioconda/bioconda-recipes/jobs/170718680#L599) occurs at build time. May have something [to do with zlib](https://travis-ci.org/bioconda/bioconda-recipes/jobs/170718680#L577).

**Comment by bgruening** [comment]
Seems to be a compilation problem, llvm seems to ignore it:
```
reader.c:715:5: error: ‘valid_format’ undeclared (first use in this function)
     valid_format = 1;
```

**Comment by marcelm** [comment]
It seems to be a bug in pear. The `valid_format` variable is only declared when the `HAVE_BZLIB_H` macro is defined, but the variable is also used in the gzip code.
```
  #ifdef HAVE_BZLIB_H
  int error;
  int valid_format = 0;
  #endif
```
The configure script does not find the bz2 library:
```
checking for library containing BZ2_bzCompress... no
```
The easiest solution would be to add a proper dependency on libbz2, but I cannot find a conda package for that anywhere (bz2file is a Python package).

**Comment by bgruening** [comment]
Try bzip2 :)

**Comment by marcelm** [comment]
Yepp, just found out! Trying now.

**Comment by marcelm** [comment]
Feel free to merge if this finishes successfully.

**Comment by bgruening** [comment]
I guess we need a rebase against master here. Sorry.

**Comment by marcelm** [comment]
Weird, the last push I made *was* already rebased on top of master. BTW, where exactly is the error [in the log](https://travis-ci.org/bioconda/bioconda-recipes/jobs/175556264)?

**Comment by bgruening** [comment]
The error is probably hidden in some more detailed log messages. Can you run it locally with --loglevel=debug?

**Comment by marcelm** [comment]
How do I do that? I tried this:
```
export TRAVIS_BUILD_DIR=/tmp/travis
./simulate-travis.py --packages pear --loglevel=debug
```
But then it complains that it cannot find `mulled-build`. Should I run the above in a docker container?

**Comment by bgruening** [comment]
You need the dependency `galaxy-lib`, not sure why, but if you do `pip install galaxy-lib` it should work.

**Comment by marcelm** [comment]
Thanks, that worked.

The error messages aren’t too helpful. First there is an `[ERROR]`, but then it says the recipe was built successfully.
```
[Nov 15 10:57:10] SERR [ERROR] - Cannot open file fwd.fastq.gz
[Nov 15 10:57:10] SERR [ERROR] - Cannot open file rev.fastq.gz
[Nov 15 10:57:10] ERRO Task processing failed: Unexpected exit code [1] of container [a14da7259d79 step-6dcb569e90], container preserved
WARNING: Overriding mulled image name, auto-detection of 'mulled' package attributes will fail to detect result.
./involucro -v=3 -f /usr/local/miniconda3/lib/python3.5/site-packages/galaxy/tools/deps/mulled/invfile.lua -set CHANNELS='bioconda,file://usr/local/miniconda3/conda-bld' -set TEST='pear -f fwd.fastq.gz -r rev.fastq.gz -o out > /dev/null && pearRM -f fwd.fastq.gz -r rev.fastq.gz -o out > /dev/null' -set TARGETS='pear=0.9.6=2' -set REPO='quay.io/mulled/tmp' -set BINDS='build/dist:/usr/local/,/usr/local/miniconda3/conda-bld:/usr/local/miniconda3/conda-bld' build-and-test

DEBUG bioconda_utils.build:build(130): STDERR:
None
INFO bioconda_utils.build:build_recipes(361): BIOCONDA BUILD SUMMARY: successfully built 1 of 1 recipes
```
The `.fastq.gz` files are used during testing, and it seems they do not get copied over to whereever they are needed for testing. They are listed in the `files:` subsection in the `test:` section, possibly there’s a bug.

I’ve rewritten the recipe such that it does not use the files, possibly that helps.

**Comment by marcelm** [comment]
Now it’s failing because of a failing `mysqlclient` recipe. How can this be? Doesn’t that mean that a broken recipe has been merged to master?

**Comment by marcelm** [comment]
... and another rebase. Feels like a wild goose chase.

**Comment by bgruening** [comment]
Sorry, the the build-system is eating some errors or at least hiding them. We will try to allocate some time to fix it :(

**Comment by marcelm** [comment]
Yes, just read the discussion on the other PRs and issues. It’s annoying, but I don’t blame anyone!

**Comment by marcelm** [comment]
Wow, incredible, it passed - after three weeks of trying :-)

**Pull request merged**